content/essentials/intro-to-reactivity.mdx

```diff
@@ -0,0 +1,171 @@
+This means that a computed value or function can intelligently determine when to recalculate based on the signals it depends on. 
+
+#### Caching
+
+This approach involves the caching of results obtained from resource-intensive function calls. 
+Instead of recalculating the output each time a function is invoked, the cached result is returned when the same inputs are encountered again. 
+
+By remembering the last computed value and the conditions (or signal values) under which it was computed, the system can avoid unnecessary recalculations.
+
+#### Dependency tracking
+
+An essential aspect of effective memoization in a reactive system is accurate dependency tracking.  
+This involves keeping track of the signals upon which a computed value relies. 
+This way, the system can identify when the cached value needs to be invalidated and recomputed. 
+
+This fine-grained dependency tracking ensures that the system's reactivity remains reliable and accurate.
+
+### Store
+
+Stores are important as they provide a structured and centralized approach to state management. 
+
+A store in reactive programming is essentially a container for state. 
+Unlike dealing with individual signals, stores offer a more organized way to manage state, especially in complex applications.
+
+#### State management
+
+At their core, stores serve as containers for state. 
+They facilitate centralized management of state by offering methods to update, reset, or manipulate it. 
+This not only improves the predictability of state transitions but simplifies the process of interacting with and modifying state.
+
+When the state within a store changes, it triggers a cascade of reactive updates throughout the system. 
+Components or functions that are subscribed to a store are promptly notified of these changes, where they are able to then respond accordingly. 
+This keeps the user interface and other system components synchronized with the underlying data.
+
+#### Scalability
+
+As applications grow in complexity, managing individual signals can become challenging. 
+Stores provide a scalable solution by allowing related state to be grouped together. 
+
+## Tracking
+
+At its core, tracking is about registering subscriptions. 
+When a signal changes, all subscribed effects are re-run. 
+This enables the automatic propagation of changes throughout an application's components.
+
+Some key points to understand about tracking:
+
+- Tracking subscriptions occur when a signal, or store value, is accessed within an effect. 
+In other words, when you read the value of a signal inside an effect, that effect becomes subscribed to that signal.
+
+- Whenever an effect is created and accesses a signal, it establishes a subscription to that signal. 
+This means that the effect will be re-executed whenever the signal changes.
+
+- Tracking is scoped to effects. 
+If you access a signal outside the context of an effect, no tracking or subscription will occur. 
```

Comment:
That's not true, memos can subscribe to signals too.

Also it may be worth explicitly saying that fundamentally there's just something that can be changed (the signal), and something that can be updated (an effect or a memo). I think it would make the system way simpler to understand because one can understand immediately that if there's not something that can be changed then the app is basically static, and if there's nothing that can be updated then the app is static just the same. Like these "tools" are here to be able to write non-static stuff, with convenience and performance.